podcasts/MultiSelectActionController.swift

```diff
@@ -272,19 +271,22 @@ class MultiSelectActionController: UIViewController, UITableViewDelegate, UITabl
         editButton.setTitleColor(AppTheme.colorForStyle(.primaryInteractive01, themeOverride: themeOverride), for: .normal)
     }
 
-    private func removeShareIfMultipleEpisodesAreSelected() {
-        if numSelectedEpisodes > 1 {
-            orderedActions = orderedActions.filter { $0 != .share }
-        }
+    private func filterUnavailableActions() {
+        let episodes = actionDelegate.multiSelectedBaseEpisodes()
+        orderedActions = orderedActions.filter { $0.isVisible(with: episodes) }
     }
 
     private func showAllActions() {
         orderedActions = originalActions
+        let shareableEpisodesCount = actionDelegate
+            .multiSelectedBaseEpisodes()
+            .filter { $0 is Episode }
+            .count
```

Comment:
We could just use allSatisfy here instead of filtering and checking the count. 

```suggestion
        let hasShareableEpisodes = actionDelegate.multiSelectedBaseEpisodes().allSatisfy({ $0 is Episode })
```

Reply:
Done in 5aa0a7a83eb3149bf3ce079e53a38790fbc1b4a6